ci/docker_ci.sh

```diff
@@ -143,8 +140,10 @@ done
 
 mkdir -p "${ENVOY_DOCKER_IMAGE_DIRECTORY}"
 ENVOY_DOCKER_TAR="${ENVOY_DOCKER_IMAGE_DIRECTORY}/envoy-docker-images.tar.xz"
-echo "Saving built images to ${ENVOY_DOCKER_TAR}."
-docker save "${IMAGES_TO_SAVE[@]}" | xz -T0 -2 >"${ENVOY_DOCKER_TAR}"
+if ! is_windows; then
+    echo "Saving built dev image to ${ENVOY_DOCKER_TAR}."
+    docker save "envoyproxy/envoy-dev:${IMAGE_NAME}" | xz -T0 -2 >"${ENVOY_DOCKER_TAR}"
```

Comment:
Don't hardcode image name, this will break the security fix repo and tagged release for sure.

Reply:
k - but i just realised that they are already broken - see #17132 so they are going to require some hardcoding/retagging of some sort

it needs to be ~hardcoded to the images that are used in the verify stage

Reply:
so checking this further - and debugging a related problem (#17133 )

looking here https://dev.azure.com/cncf/envoy/_build/results?buildId=75333&view=logs&j=f1a49366-51c0-540f-0294-f73c2bb311a9&t=7fd036c9-61ec-5b59-3645-e90950dc67a6&l=34

it seems like in release builds it creates `envoyproxy/envoy:latest` (+ for alpine) and in non-release builds it creates `envoyproxy/envoy-dev:latest` (+ for alp...)

so my thought is we can just grep for the dev version and save the correct ones

Reply:
i linked to the arm build - which we dont care about for verify - but point above still stands

Reply:
updated - should work correctly (with #17133)  in non/tagged ci